Project.Xml.Test/XmlTest.cs

```diff
@@ -0,0 +1,203 @@
+            }
+        }
+
+        [TestMethod]
+        public void Arguments_WithInvalidXMLArgument()
+        {
+            //arrange
+            string xmlFile = CreateTempFile(new string[] { "<xml>", "<orange id = 'Round Orange'/>", "<orange id = 'Naval Orange'/>", "</xml>" });
+
+            //assert
+            try
+            {
+                Assert.ThrowsException<ArgumentException>(() => Program.Main(new string[] { "xml", xmlFile, }));
+            }
+            finally
+            {
+                File.Delete(xmlFile);
+            }
+        }
+
+        [TestMethod]
+        public void Arguments_WithMissingXMLArgument() => Assert.ThrowsException<ArgumentException>(() => Program.Main(Array.Empty<string>()));
+
+        [TestMethod]
+        public void Arguments_WithTooManyArguments()
+        {
+            //arrange
+            string xmlFile = CreateTempFile(new string[] { "<xml>", "<orange id = 'Round Orange'/>", "<orange id = 'Naval Orange'/>", "</xml>" });
+
+            try
+            {
+                //act
+                Program.Main(new string[] { "-xml", xmlFile, "-test1", xmlFile, "-test2", xmlFile, });
+            }
+            catch (ArgumentException aex)
+            {
+                //assert
+                Assert.Fail(aex.Message);
+            }
+            finally
+            {
+                File.Delete(xmlFile);
+            }
+        }
+
+        [TestMethod]
+        public void Arguments_WithInvalidXMLFileName() => Assert.ThrowsException<ArgumentException>(() => Program.Main(new string[] { "-xml", "BadFile.txt"}));
+
+        [TestMethod]
+        public void File_WithValidXMLFormat()
+        {
+            //arrange
+            IFileHandling ifhXml = new XmlParser();
+
+            //assert
+            Assert.IsTrue(ifhXml.GetParsedData("<xml>,<orange id = 'Round Orange'/>,<orange id = 'Naval Orange'/>,</xml>").Length > 0);
+        }
+
+        [TestMethod]
+        public void File_WithInvalidXMLFormat()
+        {
+            //arrange
+            IFileHandling ifh = new XmlParser();
+
+            //assert
+            Assert.ThrowsException<XmlException>(() => ifh.GetParsedData("<xml>, <orange id = 'Round Orange' >, </xml>"));
+        }
+
+        [TestMethod]
+        public void File_WithValidXMLNonNumericDataSorted()
+        {
+            //arrange
+            IFileHandling ifhXml = new XmlParser();
+
+            //create xml string
+            string xml = "<xml> <orange id = 'Round Orange' /><orange id = 'Naval Orange' /><orange id = 'Blood Orange' /><orange /></xml>";
+
+            //create the expected results in the expected sort order
+            string[] expectedResults = new string[] { "Blood Orange", "Naval Orange", "Round Orange", null };
+
+            //act
+            //create the testresults sorted with nulls at the end
+            string[] testResults = Program.GetSortedData(ifhXml.GetParsedData(xml)).ToArray();
```

Comment:
`Scope`: `GetSortedData` functionality is part of the console (not the parser) and should be tested in `Project.ConsoleApp.Test` via `Program.Main()`.